src/target/riscv/riscv.h

```diff
@@ -113,6 +113,11 @@ typedef struct {
 
 #define DTM_DTMCS_VERSION_UNKNOWN ((unsigned int)-1)
 
+struct reg_name_table {
+	unsigned int num_entries;
+	const char **reg_names;
```

Comment:
The elements of the array are allocated and filled dynamically, but not changed throughout the code. There is no point in changing them. I guess this is the intention behind making it an array of constant strings, which seems perfectly fine to me.

Reply:
> The elements of the array are allocated and filled dynamically

For this exact reason, I strongly believe this should not be const. Also the need to cast away the const-ness in free() to avoid compiler warnings highlights it.